packages/devtools_app/lib/src/provider/provider_screen.dart

```diff
@@ -122,3 +137,42 @@ void showProviderErrorBanner(BuildContext context) {
         .build(context),
   );
 }
+
+class _StateInspectorSettingsDialog extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, ScopedReader watch) {
+    final theme = Theme.of(context);
+
+    return DevToolsDialog(
+      title: dialogTitleText(theme, 'State inspector configurations'),
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          InkWell(
+            onTap: () => _toggleShowInternals(context),
+            child: Row(
+              children: [
+                Checkbox(
```

Comment:
can you use a NotifierCheckbox instead of watching and reading from the context? See https://github.com/flutter/devtools/blob/8c1ab6862af372b489fb791dff995279dee303b2/packages/devtools_app/lib/src/performance/performance_screen.dart#L441 for an example

Reply:
There is no ValueListenable involved

Reply:
I was suggesting that we both use a NotifierCheckbox and move the value to be in a valuenotifier on some controller. This is the pattern we use throughout the rest of DevTools and works nicely for managing state. Not blocking though, this can be done as a cleanup.